src/components/AddSample/AddSample.jsx

```diff
@@ -0,0 +1,378 @@
+
+import { useDispatch, useSelector } from 'react-redux';
+import { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+
+// material ui imports 
+import { makeStyles } from '@material-ui/core/styles';
+import { TextField, 
+  Button, MenuItem,
+  FormHelperText, FormControl, Select, Typography } from '@material-ui/core';
+
+const useStyles = makeStyles((theme) => ({
+  inputs: {
+   margin: theme.spacing(2),
+  },
+  formControl: {
+    margin: theme.spacing(2),
+    minWidth: 180,
+  },
+}));
+
+
+function AddSample() {
+  // material ui
+  const classes = useStyles();
+
+  // set stuff so we can use them
+  const dispatch = useDispatch();
+  const history = useHistory();
+
+  // get state from redux store
+  const user = useSelector(store => store.user);
+
+  // local states for input fields
+  const [ingredientName, setName] = useState('');
+  const [lotNumber, setLotNumber] = useState('');
+  const [format, setFormat] = useState('');
+  const [ingredientAmount, setAmount] = useState('');
+  const [ingredientUnit, setIngredientUnit] = useState('');
+  const [purity, setPurity] = useState('');
+  const [dateManufactured, setDate] = useState('');
+  const [extractionMethod, setMethod] = useState('');
+  const [city, setCity] = useState('');
+  const [state, setState] = useState('');
+  const [country, setCountry] = useState('');
+  const [harvestDate, setHarvestDate] = useState('');
+  const [sustainability, setSustainability] = useState('');
+  const [cropStrain, setCropStrain] = useState('');
+
+  // functions
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    console.log("form submit")
+  };
+
+  // on shipping button click
+  const shipping = (event) => {
+    event.preventDefault();
+    console.log('shipping info');
+
+    if (
+      !ingredientName ||
+      !lotNumber ||
+      !format ||
+      !ingredientAmount ||
+      !dateManufactured ||
+      !extractionMethod
+    ) {
+      alert("Please complete required inputs");
+    } else {
+      const companyID = user.companyID;
+  
+      dispatch({
+        type: 'SEND_ORDER_INFO',
+        payload: {
+          companyID,
+          ingredientName,
+          ingredientAmount,
+          ingredientUnit,
+          format,
+          purity,
+          dateManufacture,
+          lotNumber,
+          extractionMethod,
+          city,
+          state,
+          country,
+          harvestDate,
+          cropStrain,
+          Sustainability,
+          // orderId
+        }
+      })
+    }
+  }; // end shipping
+
+  const cancel = (event) => {
+    console.log("cancel")
+    setName('');
+    setLotNumber('');
+    setFormat('');
+    setPurity('');
+    setDate('');
+    setMethod('');
+    setCity('');
+    setState('');
+    setCountry('');
+    setCropStrain('');
+    setHarvestDate('');
+    setSustainability('');
+  }; // end cancel
+
+  return (<>
```

Comment:
250 lines of JSX code here, eh? 😜 

These MUI forms can get super verbose. Definitely use whitespace and comments to break up sections, and make it easier to find them later:

```jsx
    {/* Ingredient Name */}
    <div>
        <TextField
            className={classes.inputs}
            required
            label='Ingredient Name'
            variant='filled'
            value={ingredientName} 
            type="text" 
            onChange={(event) => setName(event.target.value)}/>
          {/* <input
            value={ingredientName}
            type="text"
            placeholder="Name"
            onChange={(event) => setName(event.target.value)}/> */}
        </div>
        
        {/* Lot Numbe */}
        <div>
          <TextField
            className={classes.inputs}
            required
            label='Lot Number'
            variant='filled'
            value={lotNumber} 
            type="text" 
            onChange={(event) => setLotNumber(event.target.value)}/>
          {/* <input
            value={lotNumber}
            type="text"
            placeholder="Lot Number"
            onChange={(event) => setLotNumber(event.target.value)}
          /> */}
      </div>
```

Yes, it adds _more_ lines, but it makes it easier to scan through, and find what you need.

If you want to level up, you can DRY up your code by making some utility components, to simplify some of the more verbose MUI syntax:

```jsx
function DropDown({
  value,
  onChange,
  options,
  defaultOption,
  label,
}) {
  return (
    <FormControl variant="filled" className={classes.formControl}>
      <Select
        value={value}
        onChange={(e) => onChange(e.target.value)}
        displayEmpty
        inputProps={{ 'aria-label': 'Without label' }}>

        <MenuItem value="" disabled>
          {defaultOption}
        </MenuItem>

        {Object.entries(options).map(([value, label]) => (
          <MenuItem value={value}>{label}</MenuItem>
        ))}
      </Select>
      <FormHelperText>{label}</FormHelperText>
    </FormControl>
  )
}
```
Then we can use that like:

```jsx
<DropDown 
  label="Product Format"
  value={format}
  options={{
    'Powder': 'Powder',
    'Tincture': 'Tincture',
    'Oil': 'Oil',
    'Other': 'Other'
  }}
  defaultOption="Pick a Genre"
  onChange={val => setFormat(val)}
/>
```

It's quite a bit easier to read. And an extra benefit is that if you change the styling or whatever on those dropdown, all that code is in one place.